src/esp/physics/bullet/BulletPhysicsManager.cpp

```diff
@@ -161,6 +161,14 @@ void BulletPhysicsManager::stepPhysics(double dt) {
   int numSubStepsTaken =
       bWorld_->stepSimulation(dt, /*maxSubSteps*/ 10000, fixedTimeStep_);
   worldTime_ += numSubStepsTaken * fixedTimeStep_;
+
+  // Manually sync the motionstates for all DYNAMIC objects. KINEMATIC and
+  // STATIC sync should always be one-way.
+  for (auto& objectItr : existingObjects_) {
```

Comment:
Isn't Bullet supposed to do this automatically through the motion state interface we give it? Have you investigated why this isn't happening correctly in Bullet? This feels like a band-aid.

Reply:
Fair point. Digging deeper I did find another option. I'll open another PR to iterate on this specific issue.